parser/parse.go

```diff
@@ -85,27 +89,42 @@ func ParseDns(handle *pcap.Handle) {
 	packetSource.NoCopy = true
 
 PACKETLOOP:
-	for packet := range packetSource.Packets() {
+	for {
```

Comment:
Is there a reason why you stopped looping on the `packetSource`? It's definitely more readable, and it supports NoCopy (i.e., equivalent to the `ZeroCopyReadPacketData()` called below). See this [post](https://www.devdungeon.com/content/packet-capture-injection-and-analysis-gopacket) that discusses implementing DecodingLayer using `packetSource`.

Reply:
`Packets()` adds overhead from reading/writing a channel. This is mentioned in the [gopacket docs](https://godoc.org/github.com/google/gopacket#hdr-Reading_With_NextPacket_Function).

I tested the `packetSource` implementation mentioned in the blog post and saw a significant speed decrease in benchmarks on the 35MB pcap:
<img width="1367" alt="Screen Shot 2020-07-09 at 3 17 03 PM" src="https://user-images.githubusercontent.com/5000248/87081364-3fd18500-c1f7-11ea-972a-514640c5aa09.png">

My only concern with `ZeroCopyReadPacketData()` is a potential loss of error information since we don't get an `ErrorLayer`. I'm investigating whether the ErrorLayer errors are covered by the `ZeroCopyReadPacketData()` error handling (or if there's some other way to access it)